code/modules/mob/living/living.dm

```diff
@@ -937,9 +953,191 @@ Sorry Giacom. Please don't be mad :(
+		return
+	if(eye_blind)
+		var/blind_minimum = 0
+		if(stat == UNCONSCIOUS || (disabilities & BLIND))
+			blind_minimum = 1
+		if(client)
+			if(client.eye != src)
+				var/atom/A = client.eye
+				if(global_hud.blind in A.get_vision_impairments(src))
+					blind_minimum = 1
+		eye_blind = max(eye_blind+amount, blind_minimum)
+		if(!eye_blind)
+			clear_alert("blind")
+			update_vision_overlays()//if we're no longer blind, we update our vision overlays.
+
+
+/mob/proc/set_blindness(amount)
+	return
+
+/mob/living/set_blindness(amount)
+	if(amount>0)
+		if(eye_blind >= amount)
+			return
+		var/old_eye_blind = eye_blind
+		eye_blind = amount
+		if(client && !old_eye_blind)
+			update_vision_overlays()
+			throw_alert("blind", /obj/screen/alert/blind)
+	else if(eye_blind)
+		var/blind_minimum = 0
+		if(stat == UNCONSCIOUS || (disabilities & BLIND))
+			blind_minimum = 1
+		if(client)
+			if(client.eye != src)
+				var/atom/A = client.eye
+				if(global_hud.blind in A.get_vision_impairments(src))
+					blind_minimum = 1
+		eye_blind = blind_minimum
+		if(!eye_blind)
+			clear_alert("blind")
+			update_vision_overlays()
+
+/mob/proc/adjust_blurriness(amount)
+	if(amount>0)
+		var/old_eye_blurry = eye_blurry
+		eye_blurry += amount
+		if(!old_eye_blurry)
+			update_vision_overlays()
+	else if(eye_blurry)
+		eye_blurry = max(eye_blurry+amount, 0)
+		if(!eye_blurry)
+			update_vision_overlays()
+
+/mob/proc/set_blurriness(amount)
+	var/old_eye_blurry = eye_blurry
+	eye_blurry = amount
+	if(amount>0)
+		if(!old_eye_blurry)
+			update_vision_overlays()
+	else if(old_eye_blurry)
+		update_vision_overlays()
+
+
+/mob/proc/set_eye_stat(amount)
+	return
+
+/mob/living/carbon/set_eye_stat(amount)
+	var/old_eye_stat = eye_stat
+	if(amount>0)
+		eye_stat = amount
+		update_vision_overlays()
+	else if(old_eye_stat)
+		eye_stat = 0
+		update_vision_overlays()
+
+/mob/proc/adjust_eye_stat(amount)
+	return
+
+/mob/living/carbon/human/adjust_eye_stat(amount)
+	var/old_eye_stat = eye_stat
+	if(amount>0)
+		eye_stat += amount
+		if(!old_eye_stat)
+			update_vision_overlays()
+	else if(old_eye_stat)
+		eye_stat = max(eye_stat+amount, 0)
+		if(old_eye_stat && !eye_stat)
+			update_vision_overlays()
+
+/mob/proc/adjust_drugginess(amount)
+	return
+
+/mob/living/carbon/adjust_drugginess(amount)
+	var/old_druggy = druggy
+	if(amount>0)
+		druggy += amount
+		if(!old_druggy)
+			update_vision_overlays()
+			throw_alert("high", /obj/screen/alert/high)
+	else if(old_druggy)
+		druggy = max(eye_blurry+amount, 0)
+		if(!druggy)
+			update_vision_overlays()
+			clear_alert("high")
+
+/mob/proc/set_drugginess(amount)
+	return
+
+/mob/living/carbon/set_drugginess(amount)
+	var/old_druggy = druggy
+	druggy = amount
+	if(amount>0)
+		if(!old_druggy)
+			update_vision_overlays()
+			throw_alert("high", /obj/screen/alert/high)
+	else if(old_druggy)
+		update_vision_overlays()
+		clear_alert("high")
+
+
+/mob/proc/update_vision_overlays()

```

Comment:
The proc to handle the fullscreen visual overlays, like blind overlay or the welding helmet overlay.